wgsl/index.bs

```diff
@@ -4958,6 +4958,18 @@ The <dfn noexport>indirection</dfn> operator converts a pointer to its correspon
 
     | [=syntax/primary_expression=] [=syntax/postfix_expression=] ?
 </div>
+<div class='syntax' noexport='true'>
+  <dfn for=syntax>lhs_expression</dfn> :
+
+    | ( [=syntax/star=] | [=syntax/and=] ) * [=syntax/core_lhs_expression=] [=syntax/postfix_expression=] ?
```

Comment:
This allows a lhs_expression to be `& v` which is a pointer.  But the LHS of an assignment like this must be a reference.
So we can further restrict by making the production be:

               ( start and ) * core_lhs_expression

Reply:
I don't think `(star and) * core_lhs_expression` works, as it would forbid `*x = 42;` which I believe to be a valid assignment.

The closest I can see work is `(star | (star and) ) * core_lhs_expression` but it does not strike me as any simpler than `(star | and) * core_lhs_expression`.

And I am only restricting the syntax in the first place to prevent a conflict in the grammar (for which `(star | and) *` is already more than enough). We cannot fully prevent all non-sensical code at the grammar level and that is fine since the typing will take care of it afterwards.

Reply:
My mistake. You're right.  Thanks